base_layer/wallet/src/transaction_service/protocols/transaction_broadcast_protocol.rs

```diff
@@ -451,6 +473,14 @@ where TBackend: TransactionBackend + 'static
             },
         };
 
+        if !response.is_synced {
+            info!(
+                target: LOG_TARGET,
+                "Base Node reports not being synced, submission will be retried."
+            );
+            return Ok(false);
+        }
```

Comment:
We have a bit of a logic problem here. This does not account for the case where a transaction has been mined with the required number of confirmations, even though the base node may not be synced. It should still count as mined.

Reply:
Maybe:
``` rust
        if !response.is_synced {
            if !(response.location == TxLocation::Mined && response.confirmations >= self.resources.config.num_confirmations_required as u64) {
                info!(
                    target: LOG_TARGET,
                    "Base Node reports not being synced, submission will be retried."
                );
                return Ok((false, Some(response.height)));
            }
        }
```

Reply:
I agree, this is a safe assumption. Will update.